django/core/management/commands/squashmigrations.py

```diff
@@ -140,14 +150,37 @@ def handle(self, **options):
         # We need to take all dependencies from the first migration in the list
         # as it may be 0002 depending on 0001
         first_migration = True
+        ignore_restricted_categories = set([
+            OperationCategory.ADDITION,
+            OperationCategory.ALTERATION
+        ])
+
+
         for smigration in migrations_to_squash:
             if smigration.replaces:
                 raise CommandError(
                     "You cannot squash squashed migrations! Please transition it to a "
                     "normal migration first: https://docs.djangoproject.com/en/%s/"
                     "topics/migrations/#squashing-migrations" % get_docs_version()
                 )
+
+
+            if not first_migration and ignore_dependencies:
+                filtered_operations = []
+
+                for operation in smigration.operations:
+                    if (operation.category in ignore_restricted_categories
+                    and isinstance(operation.field, models.fields.related.RelatedField)):
+                        #and operation field app_label != app_label:
+                        continue
+
+                    filtered_operations.append(operation)
+
+                operations.extend(filtered_operations)
+                continue
+
             operations.extend(smigration.operations)
```

Comment:
This should probably be put under `else:`

Reply:
I was working through the logic at that point, prior to changing the existing code. I just committed a more complete solution. Perhaps, you could clarify something else for me, however.
1. How do I clean up the squash file my test creates?
2. Should I exclude multi-app `ModelOperation`s altogether, or mutate them to exclude fields that reference other apps?
3. What would you recommend doing in the case of swappable tuple? Is there a predictable way to check if it's referencing an external app? Let me know if the logic I currently have makes sense.

I'm still writing tests. Also, let me know if I missed anything.

PS: I may decide to check the `dependencies` for multi-app dependencies, before combing through all of the operations to find references to other apps.